sentence_transformers/evaluation/BinaryClassificationEvaluator.py

```diff
@@ -144,25 +149,31 @@ def __call__(self, model, output_path: str = None, epoch: int = -1, steps: int =
         return main_score
 
     def compute_metrices(self, model):
-        try:
-            # If the sentences are hashable, then we can use a set to avoid embedding the same sentences multiple times
-            sentences = list(set(self.sentences1 + self.sentences2))
```

Comment:
I think the intention here was to catch the `TypeError` *only* out of the set(...)` conversion. So move the rest of the `try` block here to the `else` block

Reply:
Sounds good!